larray/io/session.py

```diff
@@ -109,19 +109,28 @@ def read_arrays(self, keys, *args, **kwargs):
         self.close()
         return res
 
-    def dump_arrays(self, key_values, *args, **kwargs):
+    def dump_arrays(self, key_values, overwrite_file, *args, **kwargs):
         """
         Dumps arrays corresponds to keys in file in HDF, Excel, CSV, ... format
 
         Parameters
         ----------
         key_values : list of (str, LArray) pairs
             Name and data of arrays to dump.
+        overwrite_file : bool
+            Whether or not to overwrite the existing file (or just modify the specified sheet).
         kwargs :
             * display: whether or not to display when the dump of each array is started/done.
         """
         display = kwargs.pop('display', False)
-        self._open_for_write()
+        # for XLWingsHandler, the job is already done in the Workbook class
+        if not isinstance(self, (PandasCSVHandler, XLWingsHandler)) and overwrite_file and os.path.isfile(self.fname):
+            fname = self.fname
```

Comment:
* s/fname/final_fname/ or something similar. Having both fname and self.fname have different meaning/values is too confusing.
* when you have to check for specific backends in generic code, this smells like a bad design. I would rather move this functionality in one of the overridden methods (_open_for_write or probably a new one to avoid duplicating code). This method might even be on the generic/abstract FileHandler class, but this way each backend can override it or not.